src/Package/Impl/Publishing/Handlers/RmdPublishHandler.cs

```diff
@@ -25,12 +29,45 @@ internal sealed class RmdPublishHandler : IMarkdownFlavorPublishHandler {
             return true;
         }
 
-        public string GetCommandLine(string inputFile, string outputFilePath, PublishFormat publishFormat, Encoding encoding) {
-            string format = GetDocumentTypeString(publishFormat);
+        public async Task Publish(IRSession session, string inputFilePath, string outputFilePath, PublishFormat publishFormat, Encoding encoding) {
+            // Send the markdown to R-Host for processing as a blob
+            byte[] rmdData = File.ReadAllBytes(inputFilePath);
+            var rmdBlobId = await session.SendBlobAsync(rmdData);
```

Comment:
There will be functionality to send specific file to the remote machine, so there will be other scenarios with file reading/writing

---
In reply to: [71459410](https://github.com/Microsoft/RTVS/pull/2102#discussion_r71459410) [](ancestors = 71459410)